File: source/back-end/common/src/main/java/cn/czfshine/tiku/pojo/ProblemFullData.java
package cn.czfshine.tiku.pojo;

import cn.czfshine.tiku.pojo.dao.Answer;
import cn.czfshine.tiku.pojo.dao.Problem;
import cn.czfshine.tiku.pojo.dao.Status;
import cn.czfshine.tiku.pojo.dao.Tag;
import lombok.Data;

import java.util.HashMap;
import java.util.List;

/**
 *一个问题的内部表达形式,聚合一个问题的所有相关数据
 */

@Data
public class ProblemFullData {
    //数据库的problem表
    private Problem problem;
    //数据库的answer表
    private Answer answer;
    // 标签表
    private List<Tag> tags;
    //额外数据 ext_data表，处理成map
    private HashMap<String,String> extData;
    //状态表
    private Status status;
    /**转换成map，主要是用来返回给前台，返回的map只会（也只能）被框架解析成json对象
     * 对象的key约束和意义见开发文档
     * @return
     */
    public HashMap<String,Object> toMap(){
        return new HashMap<>();
    }
}

File: source/front-end/tiku/src/data/model/ProblemFullData.ts
interface ProblemFullData {
    /**
<<<<<<< HEAD
     * problem 问题的主体, 
     *  id： 问题的id；
     *  problemText： 问题的内容；
     *  parentId: 父问题的id
     *  answer：问题的答案id
=======
     * @problem 问题的主体:
     *  @id 问题的id;
     *  @problemText 问题的内容;
     *  @parentId 父问题的id;
     *  @answer 问题的答案id;
>>>>>>> ec267ee6943a37c04ce53aa8713730db0dd39465
     */
    /**
     * answer 问题的答案
     *  id: 标签的id
     *  value: 标签的内容
     *  parentId:  父标签的id
     */
    /**
     * tag 问题的标签
     *  id: 标签的id
     *  value: 标签的内容
     *  parentId:  父标签的id
     */
    /**
     * extData 问题的额外数据,例如: 图片、语音 等
     */

    /**
<<<<<<< HEAD
     * status 问题的状态
     */

    problem: {
        id: number,
        problemText: string,
        parentId: number,
        answerId: number
    },
    answer: any,
    tags:[{
      id: number,
      value: string,
      parentId: number
    }],
    extData: any,
    status: number
=======
    * status 问题的状态
    */
    status: any
>>>>>>> ec267ee6943a37c04ce53aa8713730db0dd39465
}
export default ProblemFullData;